### ISSUE TITLE
Format a column for one view

### ISSUE THREAD

**Issue by jennsih** [SharePoint/sp-dev-docs#5022]
I was looking for a solution that when viewer selects one of the views I've created - then one of my columns should be formatted. But all I see are examples of how to format the entire row.
If i format the column, it shows in all my views which is not what I need.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 002a1ee9-dc9f-8dc1-6f3c-ef23297b83d4
* Version Independent ID: 82565c7e-b264-94f5-8708-d4757482e133
* Content: [Use view formatting to customize SharePoint](https://docs.microsoft.com/en-us/sharepoint/dev/declarative-customization/view-formatting#feedback)
* Content Source: [docs/declarative-customization/view-formatting.md](https://github.com/SharePoint/sp-dev-docs/blob/master/docs/declarative-customization/view-formatting.md)
* Product: **sharepoint**
* GitHub Login: @spdevdocs
* Microsoft Alias: **spdevdocs**

**Comment by andrewconnell** [comment]
@thechriskent / @bcameron1231 - either if you know if list formatting supports specific views?

**Comment by bcameron1231** [comment]
Unfortunately, Column Formatting is not view specific (As OP has noticed), they are specific to the column itself. 

View Formatting is the only way to create specific formatting on views. Which wouldn't allow you to target a specific column to format.

**Comment by thechriskent** [comment]
Beau is completely right. Column formatting is applied directly to the field or site column and the only way to override it is with view formatting. View Formatting is applied directly to the views so that you can have different formats per view.

However, it is _possible_ to simulate knowledge of the view within column formatting. To be clear, your format has no access to context such as which view is selected, item count, item selection, etc. However, you can check for the presence of additional columns - that have values.

This isn't a great use case but something similar to this can be done:

```json
{
  "elmType": "div",
  "children": [
    {
      "elmType": "span",
      "txtContent": "@currentField",
      "style": {
        "display": "=if(toString([$Dog]), 'none', 'inherit')"
      }
    },
    {
      "elmType": "span",
      "txtContent": "@currentField",
      "style": {
        "color": "orange",
        "display": "=if(toString([$Dog]), 'inherit', 'none')"
      }
    }
  ]
}
```

Here is a view without the `Dog` column:
![image](https://user-images.githubusercontent.com/8364109/70934958-1e1b7100-200d-11ea-89a5-737d457e9f74.png)

Here is a view with the `Dog` column:
![image](https://user-images.githubusercontent.com/8364109/70934977-2a9fc980-200d-11ea-8bc1-095281364070.png)

So... it's something... But overall it's not great. Here are some reasons why:

- Switching between views keeps the values of the columns already retrieved. This means if you start with view 1 above you won't see orange. Switch to View 2 and you'll see orange. Yay! But switch back to view 1 and you'll still see orange because the Dog column has already been added to the data object behind the scenes and it will take a refresh to unload it.
- The reference column always has to have data. No data will indicate the same thing as the column not existing.
- There are many times views are going to have the exact same columns and you won't be able to distinguish between them

So bottom line, View Formatting is the right move when you need to vary formats between views. However, it's worth noting that you have access to the view id (legacy page context) when using a Field Customizer SPFx Extension. But this may be more than you want to tackle.
